lib/streamlit/commands/navigation.py

```diff
@@ -187,6 +187,8 @@ def navigation(
     # Ordain the page that can be called
     page_to_return._can_be_called = True
     msg.navigation.page_script_hash = page_to_return._script_hash
+    # With an identified page to run, update the current page script hash
```

Comment:
nit: maybe the comment can be simplified: `# Set the current page script hash to the page that is going to be executed`